### PR TITLE
allow removing RBD mirror from a node

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Add a playbook that lets users remove RBD mirror from a node and add a test for it.

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
jenkins test dev-centos-non_container-shrink_rbdmirror

**Comment by rishabh-d-dave** [comment]
jenkins test dev-centos-container-shrink_mds

**Comment by rishabh-d-dave** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        msg: "Exiting shrink-rbdmirror playbook, no rbd-mirror was removed.
+           To shrink the cluster, either say 'yes' on the prompt or
+           or use `-e ireallymeanit=yes` on the command line when
+           invoking the playbook"
+      when: ireallymeanit != 'yes'
+
+    - name: set_fact container_exec_cmd for mon0
+      when: containerized_deployment | bool
+      set_fact:
+        container_exec_cmd: >
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]:
```suggestion
        container_exec_cmd: {{ container_binary }} exec ceph-mon-{{ hostvars[groups[mon_group_name][0]]['ansible_hostname'] }}
```

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          {{ container_binary }} exec ceph-mon-{{ hostvars[groups
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]:
```suggestion
```

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          [mon_group_name][0]]['ansible_hostname'] }}
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]:
```suggestion
```

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    - name: exit playbook, if can not connect to the cluster
+      command: >
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]: ditto

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      register: ceph_health
+      until: ceph_health.rc
+      failed_when: ceph_health.rc != 0
+      delegate_to: "{{ groups[mon_group_name][0] }}"
+      retries: 5
+      delay: 2
+
+    - name: set_fact rbdmirror_to_kill_hostname
+      set_fact:
+        rbdmirror_to_kill_hostname: >
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]: ditto

File: infrastructure-playbooks/shrink-rbdmirror.yml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    - name: purge related directories
+      file:
+        path: /var/lib/ceph/bootstrap-rbd-mirror/{{ cluster }}-{{ rbdmirror_to_kill_hostname }}
+        state: absent
+      delegate_to: "{{ rbdmirror_to_kill }}"
+
+  post_tasks:
+    - name: show ceph health
+      command: >
```
Comment (guits) [inline infrastructure-playbooks/shrink-rbdmirror.yml]: ditto

**Comment by rishabh-d-dave** [comment]
> please, avoid using this syntax when it's not really required

Personally, I like to go with the standard: wrap text at 80. Anyways, I'll get rid if it.

**Comment by guits** [comment]
> > please, avoid using this syntax when it's not really required
> 
> Personally, I like to go with the standard: wrap text at 80. Anyways, I'll get rid if it.

IMO, wrapping text at 80 chars was useful back in the '90s, even if there are still good reasons to wrap like this, it doesn't really apply here (ceph-ansible), by the way, I think today your screen resolution is good enough to support more than 80 chars

**Comment by rishabh-d-dave** [comment]
> > > please, avoid using this syntax when it's not really required
> > 
> > 
> > Personally, I like to go with the standard: wrap text at 80. Anyways, I'll get rid if it.
> 
> IMO, wrapping text at 80 chars was useful back in the '90s, even if there are still good reasons to wrap like this, it doesn't really apply here (ceph-ansible), by the way, I think today your screen resolution is good enough to support more than 80 chars

That's true. But today, wrapping is more about reading ergonomics, AFAIK. I suppose this is why most guildlines around Python coding style and Git still recommend it. 

Anyways, it's only a convention. I am OK with whatever a project prefers.

**Comment by guits** [comment]
> > > > please, avoid using this syntax when it's not really required
> > > 
> > > 
> > > Personally, I like to go with the standard: wrap text at 80. Anyways, I'll get rid if it.
> > 
> > 
> > IMO, wrapping text at 80 chars was useful back in the '90s, even if there are still good reasons to wrap like this, it doesn't really apply here (ceph-ansible), by the way, I think today your screen resolution is good enough to support more than 80 chars
> 
> That's true. But today, wrapping is more about reading ergonomics, AFAIK. I suppose this is why most guildlines around Python coding style and Git still recommend it.

Yes, this is why I said:
> `even if there are still good reasons to wrap like this, it doesn't really apply here (ceph-ansible)`

> 
> Anyways, it's only a convention. I am OK with whatever a project prefers.

When having a really long and complex command in ceph-ansible, I agree, it's fine to wrap like this.
I think a good compromise would be to keep wrapping but only at 110/120chars, although it could be less according to the command to be wrapped (for instance, if lot of complexity).

**Comment by rishabh-d-dave** [comment]
@guits the pipeline completed running successfully.

**Comment by guits** [comment]
@rishabh-d-dave IRC? :smile:

**Comment by guits** [comment]
@rishabh-d-dave we are still missing something in this playbook I think, we must check if the daemon has been actually removed.

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-container-shrink_rbdmirror

**Comment by guits** [comment]
jenkins test dev-centos-non_container-shrink_rbdmirror

**Pull request merged**